src/figuresClasses.ts

```diff
@@ -2,10 +1,83 @@
+type Shapes = 'triangle' | 'circle' | 'rectangle';
+type Colors = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
+  getArea(): number;
+}
+
+function checkError(condition: boolean, message: string): void {
+  if (condition) {
+    throw new Error(message);
+  }
+}
+
+export class Triangle implements Figure {
+  readonly shape = 'triangle';
+
+  constructor(
+    public color: Colors,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    checkError(a <= 0 || b <= 0 || c <= 0, 'One or more length <= 0');
+
+    checkError(
+      a + b <= c || a + c <= b || b + c <= a,
+      "Your sides can't form a triangle",
+    );
+  }
+
+  getArea(): number {
+    const semiperimeter = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(
+      semiperimeter *
+        (semiperimeter - this.a) *
+        (semiperimeter - this.b) *
+        (semiperimeter - this.c),
+    );
+
+    return Math.floor(area * 100) / 100;
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  readonly shape = 'circle';
 
-export class Rectangle implements Figure {}
+  constructor(
+    public color: Colors,
+    private radius: number,
+  ) {
+    checkError(radius <= 0, 'Not corect redius');
```

Comment:
Make sure the error message is descriptive and correctly spelled. Instead of 'Not corect redius', it should be something like 'Radius must be greater than 0'.